pkg/util/cmputil/reporter.go

```diff
@@ -14,7 +14,13 @@ type DiffReport []Diff
 func (r DiffReport) GetDiffsForField(path string) DiffReport {
 	var result []Diff
 	for _, diff := range r {
-		if strings.HasPrefix(path, diff.Path) {
+		if strings.HasPrefix(diff.Path, path) {
```

Comment:
Can you add a test that explains the difference between the new and old behavior?